dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/SubsystemFeatureDefinitionImpl.java

```diff
@@ -549,27 +541,109 @@ public Collection<Bundle> getBundles() {
         return bundles;
     }
 
+    /**
+     * Tell if this is a versionless feature.
+     *
+     * Currently these are:
+     *
+     * <ul><li>public</li>
+     * <li>platformless</li>
+     * <li>have a short name that is equal to the feature name</li>
+     * <li>contain ".versionless." in their symbolic name.</li>
+     * </ul>
+     *
+     * @return True or false telling if this is a versionless feature.
+     */
     @Override
     public boolean isVersionless() {
-        // TODO Auto-generated method stub
-        return false;
+        if (!isPublic() || (getPlatformValue() != null)) {
+            return false;
+        }
+
+        String shortName = getIbmShortName();
+        if ((shortName == null) || !shortName.equals(getFeatureName())) {
+            return false;
+        }
+
+        if (getSymbolicName().indexOf(".versionless.") == -1) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Tell if this is a compatibility feature.
+     *
+     * <ul><li>private</li>
+     * <li>do not have a short name</li>
+     * <li>contain ".eeCompatible-" or ".mpCompatible-" in their symbolic name.</li>
+     * </ul>
+     *
+     * @return True or false telling if this is a versionless feature.
+     */
+    @Override
+    public boolean isCompatibility() {
+        if (!isPrivate()) {
+            return false;
+        } else if (getIbmShortName() != null) {
+            return false;
+        }
+
+        String symName = getSymbolicName();
+        return (symName.contains(".eeCompatible-") ||
+                symName.contains(".mpCompatible-"));
```

Comment:
should be `return (getPlatformValue() != null)` or something like that. They are the only private features with a wlp-platform value. Hard coding a name is unnecessary. Also, the compatibility feature for microprofile features is `mpVersion` so this would return the wrong answer for those features.